analytics/pubstack/eventchannel/sender.go

```diff
@@ -26,6 +27,7 @@ func NewHttpSender(client *http.Client, endpoint string) Sender {
 		if err != nil {
 			return err
 		}
+		defer resp.Body.Close()
```

Comment:
Does this need to be a `defer` since the function ends right after? Could it just call `resp.Body.Close()` directly? .. or does closing it immediately affect the ` resp.StatusCode`?

Reply:
yes no need of ```defer```, updated PR.